packages/contracts/src/combat/LibActiveCombat.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { ActiveCombat } from "../codegen/Tables.sol";
+
+library LibActiveCombat {
+  error LibActiveCombat__CombatActive();
+  error LibActiveCombat__CombatNotActive();
+  error LibActiveCombat__CombatActiveForDifferentRetaliator();
+
+  function getRetaliatorEntity(bytes32 initiatorEntity) internal view returns (bytes32 retaliatorEntity) {
+    if (ActiveCombat.get(initiatorEntity) != bytes32(0)) {
+      revert LibActiveCombat__CombatNotActive();
+    }
+    return ActiveCombat.get(initiatorEntity);
+  }
+
+  function requireActiveCombat(bytes32 initiatorEntity, bytes32 retaliatorEntity) internal view {
+    if (ActiveCombat.get(initiatorEntity) == bytes32(0)) {
+      revert LibActiveCombat__CombatNotActive();
+    }
+    if (ActiveCombat.get(initiatorEntity) != retaliatorEntity) {
+      revert LibActiveCombat__CombatActiveForDifferentRetaliator();
+    }
+  }
+
+  function requireNotActiveCombat(bytes32 initiatorEntity) internal view {
+    if (ActiveCombat.get(initiatorEntity) == bytes32(0)) {
+      revert LibActiveCombat__CombatActive();
+    }
+  }
```

Comment:
There're still 2 incorrect conditions (1 fixed, 1 unfixed, 1 you accidentally created)
I think it might be better if you analyze this code and find them yourself rather than me pointing it out without you understanding what's happening